pkg/gcp/client.go

```diff
@@ -64,50 +65,77 @@ type gcpClient struct {
 	serviceUsageClient         *serviceusage.Service
 }
 
+const (
+	defaultCallTimeout = 2 * time.Minute
+)
+
+func contextWithTimeout(ctx context.Context) (context.Context, context.CancelFunc) {
+	return context.WithTimeout(ctx, defaultCallTimeout)
+}
+
 func (c *gcpClient) CreateServiceAccount(ctx context.Context, request *iamadminpb.CreateServiceAccountRequest) (*iamadminpb.ServiceAccount, error) {
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
 	svcAcct, err := c.iamClient.CreateServiceAccount(ctx, request)
 	return svcAcct, err
 }
 
 func (c *gcpClient) CreateServiceAccountKey(ctx context.Context, request *iamadminpb.CreateServiceAccountKeyRequest) (*iamadminpb.ServiceAccountKey, error) {
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
 	return c.iamClient.CreateServiceAccountKey(ctx, request)
 }
 
 func (c *gcpClient) DeleteServiceAccount(ctx context.Context, request *iamadminpb.DeleteServiceAccountRequest) error {
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
 	return c.iamClient.DeleteServiceAccount(ctx, request)
 }
 
 func (c *gcpClient) DeleteServiceAccountKey(ctx context.Context, request *iamadminpb.DeleteServiceAccountKeyRequest) error {
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
 	return c.iamClient.DeleteServiceAccountKey(ctx, request)
 }
 
 func (c *gcpClient) GetRole(ctx context.Context, request *iamadminpb.GetRoleRequest) (*iamadminpb.Role, error) {
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
 	return c.iamClient.GetRole(ctx, request)
 }
 
 func (c *gcpClient) GetServiceAccount(ctx context.Context, request *iamadminpb.GetServiceAccountRequest) (*iamadminpb.ServiceAccount, error) {
-	svcAcct, err := c.iamClient.GetServiceAccount(ctx, request)
-	return svcAcct, err
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
+	return c.iamClient.GetServiceAccount(ctx, request)
 }
 
 func (c *gcpClient) ListServiceAccountKeys(ctx context.Context, request *iamadminpb.ListServiceAccountKeysRequest) (*iamadminpb.ListServiceAccountKeysResponse, error) {
+	ctx, cancel := contextWithTimeout(ctx)
+	defer cancel()
 	return c.iamClient.ListServiceAccountKeys(ctx, request)
 }
 
 func (c *gcpClient) GetProjectIamPolicy(projectName string, request *cloudresourcemanager.GetIamPolicyRequest) (*cloudresourcemanager.Policy, error) {
-	return c.cloudResourceManagerClient.Projects.GetIamPolicy(projectName, request).Do()
+	ctx, cancel := contextWithTimeout(context.Background())
```

Comment:
Background should only be called in main.. everything else should either get the context or use TODO.
see https://golang.org/pkg/context/#Background and https://golang.org/pkg/context/#TODO